sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -197,8 +172,9 @@ class NodeHttpClient implements HttpClient {
 
         Promise.all([uploadStreamDone, downloadStreamDone])
           .then(() => {
-            request.abortSignal?.removeEventListener("abort", abortListener!);
-            return;
+            if (abortListener) {
```

Comment:
curious why need this if check as `abortListener` is already checked in `if (request.abortSignal && abortListener)` above

Reply:
Because the usage happens asynchronously, `abortListener` could have been assigned back to `undefined` according to TS

Reply:
got it!